ssl/ssl_lib.c

```diff
@@ -4427,32 +4427,22 @@ int ssl_log_rsa_client_key_exchange(SSL *ssl,
                           premaster_len);
 }
 
-int ssl_log_master_secret(SSL *ssl,
-                          const uint8_t *client_random,
-                          size_t client_random_len,
-                          const uint8_t *master,
-                          size_t master_len)
-{
-    /*
-     * TLSv1.3 changes the derivation of the master secret compared to earlier
-     * TLS versions, meaning that logging it out is less useful. Instead we
-     * want to log out other secrets: specifically, the handshake and
-     * application traffic secrets. For this reason, if this function is called
-     * for TLSv1.3 we don't bother logging, and just return success
-     * immediately.
-     */
-    if (SSL_IS_TLS13(ssl)) return 1;
-
-    if (client_random_len != 32) {
-        SSLerr(SSL_F_SSL_LOG_MASTER_SECRET, ERR_R_INTERNAL_ERROR);
-        return 0;
-    }
-
-    return nss_keylog_int("CLIENT_RANDOM",
+int ssl_log_secret(SSL *ssl,
+                   const char *label,
+                   const uint8_t *client_random,
+                   size_t client_random_len,
+                   const uint8_t *secret,
+                   size_t secret_len)
+{
+    /* We assert here because this length is always statically provided, so it
+     * should never be wrong. */
+    assert(client_random_len == 32);
```

Comment:
If this is always 32 why do we bother sending it as a parameter? In any case we should use `SSL3_RANDOM_SIZE` rather than a magic number.